source/Engine.cpp

```diff
@@ -1034,11 +1039,9 @@ void Engine::Draw() const
 		MapPanel::DrawMiniMap(player, .5f * min(1.f, jumpCount / 30.f), jumpInProgress, step);
 
 	// Draw ammo status.
-	static const double ICON_SIZE = 30.;
-	static const double AMMO_WIDTH = 80.;
-	Rectangle ammoBox = hud->GetBox("ammo");
+	ammoBox = hud->GetBox("ammo");
 	// Pad the ammo list by the same amount on all four sides.
-	double ammoPad = .5 * (ammoBox.Width() - AMMO_WIDTH);
+	ammoPad = .5 * (ammoBox.Width() - AMMO_WIDTH);
```

Comment:
These 2 could be initialized in the engine constructor rather than each frame, if they're going to be promoted to static / class fields.

Reply:
Would resizing reinitialise the object? ammoBox has a fixed size but is bound to the bottom right corner so the actual values in the Rectangle can change as the window is resized.
I'll move ammoPad out of here, the dimensions of ammoBox and AMMO_WIDTH don't change once the data files are loaded.

Reply:
Ah, yes, resizing the window still needs to be handled and we should keep the read from the interface object

Reply:
This can stay as is, then?